examples/tpm2-extendpcr/extendpcr.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2018, Ian Haken. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Extends a pcr with some data.
+package main
+
+import (
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+var (
+	tpmPath = flag.String(
+		"tpm-path",
+		"/dev/tpm0",
+		"Path to the TPM device (character device or a Unix socket).")
+	pcr = flag.Int(
+		"pcr",
+		-1,
+		"Pcr to read. Must be within [0, 23].")
+	data = flag.String(
+		"data",
+		"",
+		"The hex encoded bytes with which to extend the PCR. Must not exceed 1024 bytes.")
+)
+
+func main() {
+	flag.Parse()
+	var errors []error
+	seal(&errors)
+	for _, err := range errors {
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+	}
+	if len(errors) > 0 {
+		os.Exit(1)
+	}
+}
+
+func seal(errors *[]error) {
```

Comment:
This array looks like a return value rather than an in argument. I would expect this to be

```go
func seal() (errors []error)
```

or just

```go
func seal() []error
```
and have the function allocate the array.

Reply:
I tried to go with that originally but it doesn't work with deferred function calls since they can't modify the return value (though I didn't experiment with returning a *[]error).

Reply:
You can overwrite named returns: https://play.golang.org/p/77T29F5ro93

Reply:
Neat, will fix